### DISCUSSION TITLE
xpkg.upbound.io whitelist CIDR

### DISCUSSION THREAD

**Discussion by ilyesAj** [crossplane/crossplane#5147] [Q&A]
i'm trying to run crossplane behind a corporate proxy, i'm getting the following timeout (probably corporate firewall) :
```
cannot initialize parser backend: invalid package layer: Get "https://d3qrbvrml4iuq4.cloudfront.net/docker/registry/v2/blobs/sha256/f8/f8fa0799ccb305d27c9955c8d1b85cf99f5ebfa9a2e8e861c0c5b8b29e81c067/data?Expires=1702981381&Signature=c8vs0tFrMlIbM5N91XnRIPQgFdxNh6MugBYOgGiRhGM2kmvxFtdtuHupfcV986saVZLqwKLXLa6ol6Tlj0vRY8oCIJtDIznDrRME1nUnLGHp~xBMix0cW1RfdXFxA0N-5M5MEG1lqKmzSj-zjAHGcqpHUmzdTUschTJ3AoZIfJd1tjtCfAmcqhwnOFVbTOR8N5v7JCS5p~~rbauR3j40kZEdtY4JQykdQxwE5YMIuysjH57SXaJnxHgYMosVlgeFRY7S3AlU7nGI3Uvn1Y0pwSDHiZmueCCaBxmdlZxBNslrYhXGltY5NGALXWmiVwFu9hr3zJ6pat90HMy2Gjdn~Q__&Key-Pair-Id=APKAJHZYMHUO6YRB4JIA": dial tcp 18.239.166.228:443: i/o timeout
```
but when i dig [xpkg.upbound.io](http://xpkg.upbound.io/) it's a different ip address :
```
dig xpkg.upbound.io                                                                                                    

; <<>> DiG 9.10.6 <<>> xpkg.upbound.io
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 45723
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4000
;; QUESTION SECTION:
;xpkg.upbound.io.		IN	A

;; ANSWER SECTION:
xpkg.upbound.io.	300	IN	A	34.111.161.63
```
1. so i'm really confused from where the cloudfront address is coming from ? 
i imagine it like follow i suppose: 
```
xpkg.upbound.io -> LB -> cloudfront-> docker registry
```
2.  i'm not able to find any reference of which CIDR to whitelist to avoid those timeout ?

**Comment by jastang** [comment]
hi @ilyesAj ! looking to get a bit more information on your configuration here. do you have `HTTPS_PROXY` or related environment variables set to your proxy? also, does pulling from Dockerhub work?

**Comment by ilyesAj** [comment]
Hello @jastang , yes basically all the traffic from our cluster goes through a corporate proxy with whitelist( dns+associated to IP addresses) . So the cluster is configured with http and https proxy variables(both containerd, kubectl and in the pod level) . We re able to pull images from dockerhub, and other registries like quay.io, gcr ...

**Reply by jastang** [reply]
Thanks @ilyesAj -- and did you add whitelist entries for those other registries as well? I can confirm that we do use CloudFront as a CDN and our distribution's domain is `d3qrbvrml4iuq4.cloudfront.net`. We would like to replicate this kind of setup and improve the documentation, but that might be worth a try in the meantime.

**Reply by ilyesAj** [reply]
Yes, but on each of the registries, they either provide CIDR to whitelist or all the domains and subdomains to whitelist.
I also tried offline install documentation but it didn't  work since you're asking to mount xpkg package as a volume but I m not able to download that package or find it anywhere within providers ( gcp, aws). Building from source a provider to get the xpkg file is under documented 
https://docs.crossplane.io/latest/concepts/packages/

**Reply by thephred** [reply]
The domains we can see being used for a pull are `xpkg.upbound.io` and `d3qrbvrml4iuq4.cloudfront.net`. Could you verify that with those whitelisted it works?
